service/src/main/java/bio/terra/workspace/service/iam/SamService.java

```diff
@@ -909,6 +913,24 @@ public String getOrCreatePetSaEmail(String projectId, AuthenticatedUserRequest u
     }
   }
 
+  /**
+   * Fetch credentials of a user's pet service account in a given project. This request to Sam will
+   * create the pet SA if it doesn't already exist.
+   */
+  public AuthenticatedUserRequest getOrCreatePetSaCredentials(
+      String projectId, AuthenticatedUserRequest userRequest) throws InterruptedException {
+    GoogleApi googleApi = samGoogleApi(userRequest.getRequiredToken());
+    try {
+      String petEmail = getOrCreatePetSaEmail(projectId, userRequest);
+      String petToken =
+          SamRetry.retry(() -> googleApi.getPetServiceAccountToken(projectId, PET_SA_OAUTH_SCOPES));
+      return new AuthenticatedUserRequest().email(petEmail).token(Optional.of(petToken));
```

Comment:
If there's a possibility `petToken` can be null, use `Optional.ofNullable()`

Reply:
I don't believe Sam will ever return a 200 with a null `petToken` here, it should throw an exception in any abnormal cases. 
I'm not a Sam expert though, so I added a check with a more informative message.